src/Rules/Drupal/AccessResultConditionRule.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace mglaman\PHPStanDrupal\Rules\Drupal;
+
+use Drupal\Core\Access\AccessResult;
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\ConstantScalarType;
+use PHPStan\Type\VerbosityLevel;
+
+/**
+ * @implements Rule<Node\Expr\StaticCall>
+ */
+final class AccessResultConditionRule implements Rule
+{
+
+    /** @var bool */
+    private $treatPhpDocTypesAsCertain;
+
+    /**
+     * @param bool $treatPhpDocTypesAsCertain
+     */
+    public function __construct($treatPhpDocTypesAsCertain)
+    {
+        $this->treatPhpDocTypesAsCertain = $treatPhpDocTypesAsCertain;
+    }
+
+    public function getNodeType(): string
+    {
+        return Node\Expr\StaticCall::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (!$node->name instanceof Node\Identifier) {
+            return [];
+        }
+        $methodName = $node->name->toString();
+        if (!in_array($methodName, ['allowedIf', 'forbiddenIf'], true)) {
+            return [];
+        }
+        if (!$node->class instanceof Node\Name) {
+            return [];
+        }
+        $className = $scope->resolveName($node->class);
+        if ($className !== AccessResult::class) {
+            return [];
+        }
+        $args = $node->getArgs();
+        if (count($args) === 0) {
+            return [];
+        }
+        $condition = $args[0]->value;
+        if (!$condition instanceof Node\Expr\BinaryOp\Identical && !$condition instanceof Node\Expr\BinaryOp\NotIdentical) {
+            return [];
+        }
+        $conditionType = $this->treatPhpDocTypesAsCertain ? $scope->getType($condition) : $scope->getNativeType($condition);
+        // @phpstan-ignore-next-line
+        if (!$conditionType instanceof ConstantScalarType) {
+            return [];
+        }
+        $leftType = $this->treatPhpDocTypesAsCertain ? $scope->getType($condition->left) : $scope->getNativeType($condition->left);
+        $rightType = $this->treatPhpDocTypesAsCertain ? $scope->getType($condition->right) : $scope->getNativeType($condition->right);
+
+        if ($conditionType->getValue() === false) {
+            return [
+                RuleErrorBuilder::message(sprintf(
+                    'Strict comparison using %s between %s and %s will always evaluate to false.',
+                    $condition->getOperatorSigil(),
+                    $leftType->describe(VerbosityLevel::value()),
+                    $rightType->describe(VerbosityLevel::value()),
+                ))->identifier(sprintf('%s.alwaysFalse', $condition instanceof Node\Expr\BinaryOp\Identical ? 'identical' : 'notIdentical'))->build(),
+            ];
+        }
+        return [];
+    }
```

Comment:
See `reportAlwaysTrueInLastCondition` in https://github.com/phpstan/phpstan-src/blob/f1cf3bb34d949f01266c9eb08097144b262f4e4a/src/Rules/Comparison/StrictComparisonOfDifferentTypesRule.php#L69-L97

except we probably always want to evaluate the last condition, because there isn't a chain. `reportAlwaysTrueInLastCondition` is false by default. So I'm not sure how to handle "if always true". I think.